### PR TITLE
[UXENG-74] Added built-in sorting to tables

### PR DESCRIPTION
**Pull request by Elepert**
## Summary

Added ability to sort table columns to Table component.
Screen Shot 2020-06-30 at 10 44 52 AM

## Test Plan

Tested in storybook and added automated tests in `Table/tests/`.

## JIRA

[UXENG-74](https://optimizely.atlassian.net/browse/UXENG-74)

### CODE REVIEW

File: src/components/Table/TH/index.js [hunk 1]
```diff
@@ -51,6 +89,26 @@ TH.propTypes = {
 TH.defaultProps = {
   isCollapsed: false,
   isNumerical: false,
+  sorting: { canSort: false},
```
Comment (spham1412) [inline src/components/Table/TH/index.js:92]: missing a space?

File: src/components/Table/TH/index.js [hunk 2]
```diff
@@ -51,6 +89,26 @@ TH.propTypes = {
...
+   *  @returns {Error} Error or null
+   */
+  sorting: function verifyHandleSortExists(props) {
+    if (props.sorting.canSort && !props.sorting.handleSort) {
+      return new Error(
+        'Must pass a sorting function.'
+      );
+    } else if (props.sorting.canSort && !props.sorting.order) {
+      return new Error(
+        'Must pass a sorting direction.'
```
Comment (spham1412) [inline src/components/Table/TH/index.js:107]: Why not default to `ASC`?

Comment (Elepert) [reply src/components/Table/TH/index.js:107]: This is up to the dev to update as the sorting order is flipped around (so when the user presses the arrow, the dev needs to have a function that changes this from 'asc' to 'desc' and back and forth). If we default to 'asc' then that means that the dev doesn't have a way to update it so the sorting arrow will always point up.

**Comment by ayeshakmaz** [comment]
@Elepert Just one thing to note looking at the storybook - there seems to be weird behavior on the first click of the header, like for Status, for first click sorts properly, but the arrow doesn't change. Then the second click changes just the arrow, then the 3rd click gets it going in the right toggle state. Also the Rate sorting seems not right?

**Comment by Elepert** [comment]
The reason the Rate sorting seems weird is that the values are strings so it's sorting not numerically but via string. In that case, it compares the first element, then the second, etc.. versus the plain numerical value.

**Pull request merged**